mypy/checkexpr.py

```diff
@@ -2815,13 +2807,11 @@ def check_boolean_op(self, e: OpExpr, context: Context) -> Type:
         elif e.right_always:
             left_map = None
 
-        if right_map is None:
-            self.msg.disable_errors()
-        try:
+        # If right_map is None then we know mypy considers the right branch
```

Comment:
Oops, the original comment added in https://github.com/python/mypy/pull/3666 got shifted around over the years.